integration-tests/src/test/java/org/zowe/apiml/integration/ha/CachingService.java

```diff
@@ -0,0 +1,226 @@
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.integration.ha;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.restassured.RestAssured;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.zowe.apiml.util.categories.ChaoticHATest;
+import org.zowe.apiml.util.config.*;
+import org.zowe.apiml.util.requests.Endpoints;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.core.Is.is;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+import static org.zowe.apiml.util.config.ConfigReader.environmentConfiguration;
+
+@Slf4j
+@ChaoticHATest
+@TestInstance(TestInstance.Lifecycle. PER_CLASS)
+public class CachingService {
+
+    private static final String SERVICE = "service";
+    private static final String KEY = "aCacheKey" + new Random().nextInt();
+    private static final String VALUE = "aCacheValue";
+    private static final String MAP = "aMap";
+    private static final String MAP_KEY = "aMapCacheKey" + new Random().nextInt();
+    private static final String MAP_VALUE = "aMapCacheValue";
+    private static final String DN = "CN=Zowe Service, OU=API Mediation Layer, O=Zowe Sample, L=Prague, ST=Prague, C=CZ";
+
+    private static final KeyValue KEY_VALUE = new KeyValue(KEY, VALUE);
+    private static final KeyValue MAP_KEY_VALUE = new KeyValue(MAP_KEY, MAP_VALUE);
+
+    private Credentials credentials;
+    private List<String> baseUrls;
+
+    private boolean isUp(int index) {
+        try {
+            String url = String.format("%s/cachingservice%s", baseUrls.get(index), Endpoints.HEALTH);
+            log.info("Check if {}. Caching Service is up: {}", index + 1, url);
+
+            given()
+                .contentType(JSON)
+                .auth()
+                .basic(credentials.getUser(), credentials.getPassword())
+            .when()
+                .get(url)
+            .then()
+                .statusCode(200)
+                .body("status", Matchers.is("UP"));
```

Comment:
just because some people are using automatic formatters:

```suggestion
            //@formatter:off
            given()
                .contentType(JSON)
                .auth()
                .basic(credentials.getUser(), credentials.getPassword())
            .when()
                .get(url)
            .then()
                .statusCode(200)
                .body("status", Matchers.is("UP"));
            //@formatter:on
```

Reply:
You can apply the same to the other blocks

Reply:
done